core/src/main/scala/org/apache/spark/deploy/RRunner.scala

```diff
@@ -40,7 +40,13 @@ object RRunner {
 
     // Time to wait for SparkR backend to initialize in seconds
     val backendTimeout = sys.env.getOrElse("SPARKR_BACKEND_TIMEOUT", "120").toInt
-    val rCommand = "Rscript"
+    val rCommand = {
+      var cmd = sys.props.getOrElse("spark.sparkr.r.command", "Rscript")
+      if (sys.props("spark.submit.deployMode") == "client") {
```

Comment:
Can this actually happen ? I thought we only used RRunner when we launched non-shell jobs (i.e. when we are executing scripts).